core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/store/dot-categories-list-store.ts

```diff
@@ -0,0 +1,173 @@
+            tableColumns,
+            selectedCategories,
+            categoryBreadCrumb,
+            breadCrumbStarterIcon,
+            paginationPerPage
+        }: DotCategoriesListState) => {
+            return {
+                getCategoryEndPoint,
+                categoriesBulkActions,
+                addToBundleIdentifier,
+                tableColumns,
+                selectedCategories,
+                categoryBreadCrumb,
+                breadCrumbStarterIcon,
+                paginationPerPage
+            };
+        }
+    );
+
+    readonly categoryBreadCrumbSelector$ = this.select(
+        ({ categoryBreadCrumb }: DotCategoriesListState) => {
+            return {
+                categoryBreadCrumb
+            };
+        }
+    );
+
+    readonly updateListing = this.updater<DotListingDataTableComponent>(
+        (state: DotCategoriesListState, listing: DotListingDataTableComponent) => {
+            return {
+                ...state,
+                listing
+            };
+        }
+    );
+
+    readonly addCategoriesBreadCrumb = this.updater<MenuItem>(
+        (state: DotCategoriesListState, categoryBreadCrumb: MenuItem) => {
+            return {
+                ...state,
+                categoryBreadCrumb: [
+                    ...state.categoryBreadCrumb,
+                    { ...categoryBreadCrumb, tabindex: state.categoryBreadCrumb.length.toString() }
+                ]
+            };
+        }
+    );
+
+    readonly updateCategoriesBreadCrumb = this.updater<MenuItem[]>(
+        (state: DotCategoriesListState, categoryBreadCrumb: MenuItem[]) => {
+            return {
+                ...state,
+                categoryBreadCrumb
+            };
+        }
+    );
+
+    readonly updateSelectedCategories = this.updater<DotCategory[]>(
+        (state: DotCategoriesListState, selectedCategories: DotCategory[]) => {
+            return {
+                ...state,
+                selectedCategories: selectedCategories
+            };
+        }
+    );
+
+    readonly updateCategoryEndPoint = this.updater<string>(
+        (state: DotCategoriesListState, getCategoryEndPoint: string) => {
+            return {
+                ...state,
+                getCategoryEndPoint
+            };
+        }
+    );
+
+    private getCategoriesActions() {
+        return [
+            {
+                label: this.dotMessageService.get('Add')
+            },
+            {
+                label: this.dotMessageService.get('Delete')
+            },
+            {
+                label: this.dotMessageService.get('Import')
+            },
+            {
+                label: this.dotMessageService.get('Export')
+            }
+        ];
+    }
+
+    private getCategoryBreadCrumbs() {
+        return [
+            {
+                label: 'Top',
+                tabindex: '0'
+            }
+        ];
+    }
+
+    private getCategoriesColumns(): DataTableColumn[] {
```

Comment:
doc

Reply:
Use Document This extension to generate the docs.